7task7/src/AdList.js

```diff
@@ -0,0 +1,423 @@
+
+
+    getPage(skip, top, filterConfig) {
+        let result;
+        if (!filterConfig) {
+            result = this._items;
+        }
+
+        if (filterConfig) {
+
+            if (filterConfig.vendor) {
+                result = this._items.filter(function (item) {
+                        return item.vendor.toLowerCase().indexOf(filterConfig.vendor.toLowerCase()) + 1;
+                    }
+                );
+            }
+
+            if (filterConfig.createdAt) {
+                result = this._items.filter(function (item) {
+                    return item.createdAt >= filterConfig.createdAt;
+                });
+            }
+
+            if (filterConfig.validUntil) {
+                result = this._items.filter(function (item) {
+                    return item.validUntil.getTime() <= filterConfig.validUntil.getTime();
+                });
+            }
+
+
+            if (filterConfig.hashTagSearch && filterConfig.hashTagSearch.length!=0) {
+                result = this._items.filter(function (post) {
+                    if(typeof post.hashTags!== "undefined") {
+                        return filterConfig.hashTagSearch.every(function (tag) {
+                            return post.hashTags.includes(tag);
+                        });
+                    }
+
+                });
+            }
+        }
+        return result.sort((function(a, b) {
+                return a.createdAt-b.createdAt
+        }
+        )).slice(skip, skip+top);
+    }
+
+
+    add(item) {
+        const date = new Date();
+        item.id=+date;
+        item.createdAt = new Date();
+        if (AdList.validateAd(item)) {
+            this._items.push(item);
+            return true;
+        }
+        return false;
+    }
+
+
+    edit(id, adItem) {
+        let num = this._items.findIndex(item => item.id === id);
+
+        if (adItem.description) {
+            this._items[num].description = adItem.description;
+        }
+        if (adItem.link) {
+            this._items[num].link = adItem.link;
+        }
+        if (adItem.discount) {
+            this._items[num].discount = adItem.discount;
+        }
+        if (adItem.validUntil) {
+            this._items[num].validUntil = adItem.validUntil;
+        }
+        if (adItem.hashTags) {
+            this._items[num].hashTags = adItem.hashTags;
+        }
+        if (AdList.validateAd(this._items[num]) === false){
+            return false;
+        }
+        else
+            return true;
+    }
+
+
+    remove(id) {
+        let tmp = this._items.slice();
+        tmp.splice(tmp.findIndex(item => item.id === id), 1);
+        this._items=tmp.slice();
+        return true;
+
+    }
+
+
+    addAll(mass) {
+        let result = [];
+        for (let item of this._items) {
+            if (AdList.validateAd(item)) {
+                mass.push(item);
+            }
+            else {
+                result.push(item);
+            }
+        }
+        return result;
```

Comment:
You need to add items from `mass` to `this._items`, but I see you did vice versa.